openapi.yaml

```diff
@@ -1023,6 +1055,74 @@ components:
                   description: Job reason code
                   example: ABCDE
                   nullable: true
+    DeductionsResource:
+      type: object
+      properties:
+        id:
+          type: string
+          description: OSU ID of a employee
+          pattern: ^\d{9}$
+        type:
+          $ref: '#/components/schemas/DeductionsType'
+        links:
+          $ref: '#/components/schemas/SelfLink'
+        attributes:
+          properties:
+            responseMsg:
+              type: string
+              description: A payroll deductions success or failure message.
+              example: SUCCESS
+    DeductionsResult:
+      type: object
+      properties:
+        links:
+          $ref: '#/components/schemas/SelfLink'
+        data:
+          $ref: '#/components/schemas/DeductionsResource'
+    DeductionsPostBody:
+      type: object
+      required:
+        - data
+      properties:
+        data:
+          type: object
+          required:
+            - type
+          additionalProperties: false
+          properties:
+            type:
+              $ref: '#/components/schemas/DeductionsType'
+            attributes:
+              type: object
+              additionalProperties: false
+              required:
+                - osuId
+                - effectiveDate
+                - position
+                - suffix
+              properties:
+                osuId:
+                  type: string
+                  description: OSU ID of an employee
+                  example: '123456789'
+                  pattern: ^\d{9}$
+                  nullable: false
+                effectiveDate:
+                  type: string
+                  format: date
+                  description: ServiceNow job effective date. Format is 'YYYY-MM-DD'.
```

Comment:
Job?

Reply:
Technically it's the job effective date, which informs the deduction dates. But I can change it to deduction effective date if that's less confusing.

Reply:
Doesn't the job need to exist before creating deductions? If so, should we allow a user to enter the job effective date instead of looking it up from the database?

Reply:
Technically yes, but I need to ensure that the job from the onboarding case (and the job start date/deduction effective date) are correct in Banner and mirror the onboarding case before applying deductions. There are a lot of touch points and possibility for manual changes in Banner or ServiceNow that could throw things out of sync during the onboarding process. The consequences of adding deductions prematurely or incorrectly are worth this extra fail safe in my opinion. Also an employee can have multiple active jobs and concurrent onboarding cases opened. This ensures that we're applying deductions for the intended active position.